### DISCUSSION TITLE
How to make pyright complete a dict>

### DISCUSSION THREAD

**Discussion by cathaysia** [microsoft/pyright#3375] [Q&A] [answered]
I have a function like this:

```python
def get_pictures()->dict:
        return {
                'items': [OnlineIterm(i,self.req) for i in pageInfo['list']],
                'has_more': pageInfo['has_more'],
                'cursor': pageInfo['cursor'],
        }
```

As you see, the result of get_pictures has exactly type:

{
   "items": List[OnlineIterm[,
   "has_more": bool,
   "cursor": str
}

but when I use it:
![图片](https://user-images.githubusercontent.com/42114817/164957917-b9cc9550-98ea-462b-a893-16b1705c7754.png)

as you see, the type of pic is unknown. How do I make it's type be List[OnlineIterm] ?

**Comment by erictraut** [comment, marked as answer]
The return type of `get_pictures` is annotated as `dict`. The `dict` class is a generic class that accepts two type arguments, but you have provided neither of them in the type annotation. When type arguments are omitted, PEP 484 indicates that type checkers should assume that they are `Any`. Pyright distinguishes between explicit `Any` and implicit `Any` with the latter form referred to as `Unknown`. So you have effectively declared that the return type of `get_pictures` is `dict[Unknown, Unknown]` — i.e. a dictionary with keys and values of any type. 

In this case, you could declare the return type as `dict[str, Any]` or `dict[str, list[Sequence[OnlineIterm] | bool | str]`.

If you want your dictionaries to have a different type for each specified key value, you should define a `TypedDict` class as documented in [PEP 589](https://peps.python.org/pep-0589/).

**Reply by cathaysia** [reply]
Thanks for your reply. :)
